topics/wallet-standard.md

```diff
@@ -0,0 +1,248 @@
+# Wallet Standard
+
+## Messages
+
+
+
+### Permission
+
+The purpose of the permission messages is to establish a connection between a relying party and a wallet.
+
+#### Types
+
+- `text`: A plain `string` value.
+- `blob`: A `string` value describing binary data encoded in base64.
+
+#### Request
+
+`version` (`text`): The version of the standard used.
+
+`senderId` (`text`): The ID of the sender. This should be used to identify the sender of the message.
+
+`appMetadata`: Information about the relying party.
+- `name` (`text`): The user-friendly name of the relying party. 
+- `icon` (`text`, optional): An optional URL pointing to an icon resource representing the relying party.
+
+`networks`: A list of networks on which the relying party plans to operate.
+- `chainId`(`text`): The chain ID of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes the relying party requires. Possible values: 
+- `"canister_call"`
+
+`challenge` (`blob`): A challenge used for the wallet to sign in order to prove its access to the identity.
+
+#### Response
+
+`version` (`text`): The version of the standard used.
+
+`senderId` (`text`): The ID of the sender. This should be used to identify the sender of the message.
+
+`appMetadata`: Information about the wallet.
+- `name` (`text`): The user-friendly name of the wallet.
+- `icon` (`text`, optional): An optional URL poiting to an icon resource representing the wallet.
+
+`networks`: A list of networks on which the user has agreed the relying party can operate. This should be a subset of the `networks` from the original request.
+- `chainId` (`text`): The chain ID of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes that the user has agreed the relying party can be granted. This should be a subset of the `scopes` field from the original request. Possible values:
+- `"canister_call"`
+
+`principal` (`blob`): The principal associated with the identity the user has selected.
+
+`ledger` (optional): Ledger data.
+- `subaccount` (`blob`, optional)
+
+`challenge` (`blob`): A challenge the wallet had to sign.
+
+`signature` (`blob`): The signature of the challenge generated by the wallet.
```

Comment:
I think we need some more details here: i.e. what signature schemes are allowed, how to validate the challenge, and how it relates to the principal.

And maybe some additional guidance on how a challenge should be chosen, etc.